js/test/Ice/idleTimeout/Client.ts

```diff
@@ -0,0 +1,137 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import { Ice } from "ice";
+import { Test } from "./Test.js";
+import { TestHelper } from "../../Common/TestHelper.js";
+
+const test = TestHelper.test;
+
+// We verify that the idle check aborts the connection when the connection (here client connection) remains idle for
+// longer than idle timeout. Here, the client has an idle timeout of 1s and idle checks enabled. We intentionally
+// misconfigure the server with an idle timeout of 3s to send heartbeats every 1.5s, which is too long to prevent
+// the client from aborting the connection.
+async function testConnectionAbortedByIdleCheck(
```

Comment:
It would be good to add a test `testConnectionNotAbortedByIdleCheck` - this seems the whole point of adding the callback.
Basically:
 - no idle timeout mismatch (1s on both sides)
 - client (not server) is busy dispatching (where dispatch = sleep)
 - and yes, client keeps reading heartbeats and as a result the idle check never runs